src/chocolatey/infrastructure.app/services/NugetService.cs

```diff
@@ -1045,12 +1045,45 @@ private void set_package_names_if_all_is_specified(ChocolateyConfiguration confi
                 config.Input = string.Empty;
 
                 var localPackages = list_run(config, logResults: false);
+                var packagesToUpdate = localPackages.Select(p => p.Key).ToList();
+
+                if (!String.IsNullOrEmpty(config.UpgradeCommand.PackageNamesToSkip))
+                {
+                    var packagesToSkip = config.UpgradeCommand.PackageNamesToSkip
+                        .Split(',')
+                        .Select(x => x.Trim())
+                        .Where(x => !String.IsNullOrEmpty(x))

```

Comment:
Since it follows a `Trim`, it can never be whitespace. It cannot be `null` either... I'll simply move that line before the `trim_safe` :)